src/lib.rs

```diff
@@ -173,10 +176,54 @@ impl<T> CircularQueue<T> {
         } else {
             self.data[self.insertion_index] = x;
         }
-
+        if self.reverse_idx > 0 {
+            self.reverse_idx -= 1;
+        }
         self.insertion_index = (self.insertion_index + 1) % self.capacity();
     }
 
+    /// Pops the elements from the queue(LIFO).
+    ///
+    /// Pops at max self.capacity number of items.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use circular_queue::CircularQueue;
+    ///
+    /// let mut queue = CircularQueue::with_capacity(3);
+    /// queue.push(1);
+    /// queue.push(2);
+    /// queue.push(3);
+    /// queue.push(4);
+    ///
+    /// assert_eq!(queue.len(), 3);
+    ///
+    /// assert_eq!(queue.pop(), Some(4));
+    /// assert_eq!(queue.pop(), Some(3));
+    /// assert_eq!(queue.pop(), Some(2));
+    /// assert_eq!(queue.pop(), None);
+    /// 
+    /// ```
+    pub fn pop(&mut self) -> Option<T>
+        where T: Copy
+    {
+        if self.data.len() == 0 {
+            None
```

Comment:
This branch isn't needed because the following branch already handles `len() == 0` (inside `Vec::pop()`).